plugin/pkg/auth/authenticator/password/keystone/keystone.go

```diff
@@ -39,11 +39,30 @@ type KeystoneAuthenticator struct {
 
 // AuthenticatePassword checks the username, password via keystone call
 func (keystoneAuthenticator *KeystoneAuthenticator) AuthenticatePassword(username string, password string) (user.Info, bool, error) {
-	opts := gophercloud.AuthOptions{
+	var opts gophercloud.AuthOptions
+	opts = gophercloud.AuthOptions{
 		IdentityEndpoint: keystoneAuthenticator.authURL,
 		Username:         username,
 		Password:         password,
 	}
+	if strings.HasSuffix(keystoneAuthenticator.authURL, "/v3") {
+		// username has a format of domain\name.
+		firstIndex := strings.Index(username, "\\")
```

Comment:
> I think it's a very good improvement to support v3 for all the usernames except those with a backslash.

I'm really hesitant to introduce custom domain encoding schemes to server-side password auth. Letting the client obtain a token directly and present it to the API would be preferable to expanding the username/password support.